src/tree.c

```diff
@@ -293,22 +293,43 @@ static void ofi_delete_rebalance(struct ofi_rbmap *map, struct ofi_rbnode *node)
 	node->color = BLACK;
 }
 
+static void ofi_rbmap_exchange(struct ofi_rbmap *map,
+		struct ofi_rbnode *cur, struct ofi_rbnode *successor)
+{
+	if (successor == cur)
+		return;
+
+	*successor = *cur;
+
+	if (!cur->parent)
+		map->root = successor;
+	else if (cur == cur->parent->left)
+		cur->parent->left = successor;
+	else
+		cur->parent->right = successor;
+
+	if (cur->left != &map->sentinel)
+		cur->left->parent = successor;
+	if (cur->right != &map->sentinel)
+		cur->right->parent = successor;
+}
```

Comment:
nit: how about bringing the `free(node)` from caller to here and renaming the function to `ofi_rbmap_replace_node_ptr` and the args as `old` and `new`? (I know the function name is longer but I think that conveys the meaning better).

Reply:
I prefer the free remain in the delete call.  This is really just replacing one node with another, but your name suggestion works for that.  New as an arg name doesn't work, since it's reserved in C++ if someone wants to combine libfabric source with their project.  I actually thought about that, and went with current/successor instead (over src/dst too).

Reply:
I suggested moving the free here to make this call self-contained but it's probably not a big deal. The arg can be named as `new_`. `successor` can be misinterpreted as successor the node `cur` in the tree when considering the sorted order of the keys.

Reply:
Updated with new name, and used old_node, new_node.  You're right, successor is confusing.  I kept the free outside the call, so the new call just replaces the nodes.